### PR TITLE
Use manual trigger to prepare release PRs

### LINKED ISSUE #7507: Use new GH actions manual triggers for preparing releases
As seen in: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch

It should be simple to adapt our current script to use this new mechanism instead.

### PR DESCRIPTION
**Pull request by nicoddemus**
Unfortunately this is untested because the workflow must first be merged
to master in order to be available for triggering.

Fix #7507

I intend to leave the other workflow in place until we can test this workflow and are satisfied with it.

### CODE REVIEW

File: scripts/prepare-release-pr.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    import github3
+
+    github = github3.login(token=token)
+    owner, repo = SLUG.split("/")
+    return github.repository(owner, repo)
+
+
+def prepare_release_pr(base_branch: str, is_major: bool, token: str) -> None:
+    print()
+    print(f"Precessing release for branch {Fore.CYAN}{base_branch}")
```
Comment (bluetech) [inline scripts/prepare-release-pr.py:53]:
```suggestion
    print(f"Processing release for branch {Fore.CYAN}{base_branch}")
```

File: scripts/prepare-release-pr.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
+"""
+This script is part of the pytest release process which is manually in the Actions
```
Comment (bluetech) [inline scripts/prepare-release-pr.py:2]:
```suggestion
This script is part of the pytest release process which is triggered manually in the Actions
```

File: scripts/prepare-release-pr.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    valid_versions.sort()
+    last_version = valid_versions[-1]
+
+    changelog = Path("changelog")
+
+    features = list(changelog.glob("*.feature.rst"))
+    breaking = list(changelog.glob("*.breaking.rst"))
+    is_feature_release = features or breaking
+
+    if is_feature_release and base_branch != "master":
```
Comment (bluetech) [inline scripts/prepare-release-pr.py:131]: BTW, our current `RELEASING.md` says that also for feature releases, we first branch and make the release from the branch, which conflicts with this. I don't really remember anymore when it got out of sync. (But this is pre-existing, not directly related to this PR)

**Review by bluetech — APPROVED** [review]

File: scripts/prepare-release-pr.py (hunk 3 above)
Comment (nicoddemus) [reply scripts/prepare-release-pr.py:131]: Indeed. But I will fix this now, it is just a matter of removing the check. 👍

**Pull request merged**